gldas/interface.py

```diff
@@ -17,7 +17,109 @@
 
 from gldas.grid import GLDAS025Cellgrid
+from netCDF4 import Dataset
 
+class GLDAS_Noah_v21_025Img(ImageBase):
+    """
+    Class for reading one GLDAS Noah v2.1 nc file in 0.25° grid.
+
+    Parameters
+    ----------
+    filename: string
+        filename of the GLDAS nc file
+    mode: string, optional
+        mode of opening the file, only 'r' is implemented at the moment
+    parameter : string or list, optional
+        one or list of parameters to read, see GLDAS v2.1 documentation for more information
+        Default : 'SoilMoi0_10cm_inst'
+    array_1D: boolean, optional
+        if set then the data is read into 1D arrays. Needed for some legacy code.
+    """
+    
+    def __init__(self, filename, mode='r', parameter='SoilMoi0_10cm_inst', array_1D=False):
+        super(GLDAS_Noah_v21_025Img, self).__init__(filename, mode=mode)
+
+        if type(parameter) != list:
+            parameter = [parameter]
+        self.parameters = parameter
+        self.fill_values = np.repeat(9999., 1440 * 120)
+        self.grid = GLDAS025Cellgrid()
+        self.array_1D = array_1D
+    
+    def read(self, timestamp=None):
+        
+        #print 'read file: %s' %self.filename
+        #Returns the selected parameters for a gldas image and according metadata
+        return_img = {}
+        return_metadata = {}
+        
+        try:
+            dataset = Dataset(self.filename)
+        except IOError as e:
+            print(e)
+            print(" ".join([self.filename, "can not be opened"]))
+            raise e
+         
+
+        param_names=[]
+        for parameter in self.parameters:
+            param_names.append(parameter)
+            
+        
+        for parameter, variable in dataset.variables.items():
+            if parameter in param_names:
+                param_metadata={}
+                param_data={}           
+                for attrname in variable.ncattrs():
+                    if attrname in ['long_name','units']:
+                        param_metadata.update({str(attrname):getattr(variable,attrname)})
+                        
+                   
+                param_data=dataset.variables[parameter][:]
+                np.ma.set_fill_value(param_data,9999)  
+                
+                param_data=np.concatenate((self.fill_values,
+                                           np.ma.getdata(param_data.filled())
+                                           .flatten()))
+                                        
+                return_img.update({str(parameter):param_data[self.grid.activegpis]})
+                return_metadata.update({str(parameter):param_metadata})
+                        
+            #Check for corrupt files
+                try:
+                    return_img[parameter]
+                except KeyError:
+                    path, thefile = os.path.split(self.filename)
+                    print '%s in %s is corrupt - filling image with NaN values' %(parameter,thefile)
```

Comment:
Please change the print here to make it work with Python 3. See the failing tests.